arrayTransformations.mjs

```diff
@@ -108,104 +109,104 @@ export function scaleToRange(inputArray, inputMin, inputMax, outputMin, outputMa
 }
 
 /*
-  * scales the elements in the "vals" array proportionally based on the desired "span".
-  * @param {number} span - The desired sum to scale the values to.
-  * @param {number[]} vals - An array of numerical values to be scaled.
-  * @example console.log(scaleToSum(100, [10,20,30,40,50])) //[6.666666666666667, 13.333333333333334, 20, 26.666666666666668, 33.333333333333336]
-**/
+ * scales the elements in the "vals" array proportionally based on the desired "span".
+ * @param {number} span - The desired sum to scale the values to.
+ * @param {number[]} vals - An array of numerical values to be scaled.
+ * @example console.log(scaleToSum(100, [10,20,30,40,50])) //[6.666666666666667, 13.333333333333334, 20, 26.666666666666668, 33.333333333333336]
+ **/
 export function scaleToSum(span, vals) {
-    let inputSum = this.sum(vals);
+    let inputSum = sum(vals);
     return vals.map((x) => (x * span) / inputSum);
 }
 
 /**
-  * Picks an item randomly in an array.
-  * @param {array} inputArray - the array to pick an item from
-  * @example 
-  * console.log(pick([1,2,3,4])) //4
-  * console.log(pick([1,2,3,4])) //1
-  * console.log(pick([1,2,3,4])) //2
-  * console.log(pick([1,2,3,4])) //3
-*/
+ * Picks an item randomly in an array.
+ * @param {array} inputArray - the array to pick an item from
+ * @example
+ * console.log(pick([1,2,3,4])) //4
+ * console.log(pick([1,2,3,4])) //1
+ * console.log(pick([1,2,3,4])) //2
+ * console.log(pick([1,2,3,4])) //3
+ */
 export function pick(inputArray) {
     return inputArray[Math.round((inputArray.length - 1) * Math.random())];
 }
 
 /**
-  * Picks multiple items randomly in an array.
-  * @param {number} n - the desired number of item that are picked
-  * @param {array} inputArray - the array to pick items from
-  * @example console.log(pickN(3, [1,2,3,4,5,6,7,8,9])) //[9, 5, 8]
-*/
+ * Picks multiple items randomly in an array.
+ * @param {number} n - the desired number of item that are picked
+ * @param {array} inputArray - the array to pick items from
+ * @example console.log(pickN(3, [1,2,3,4,5,6,7,8,9])) //[9, 5, 8]
+ */
 export function pickN(n, inputArray) {
-    return this.buildArray(n, (i) => pick(inputArray));
+    return buildArray(n, (i) => pick(inputArray));
```

Comment:
In JavaScript, `this` is the parent of the instance. In this library's case, every function is top level.  These functions don't have parent instance.